packages/oidc/src/token.ts

```diff
@@ -0,0 +1,226 @@
+interface TokenPayload {
+  sub: string;
+  name: string;
+  exp: number;
+}
+
+function isNodeEnv(): boolean {
+  return (
+    typeof process !== 'undefined' &&
+    process.versions != null &&
+    process.versions.node != null
+  );
+}
+
+function isBrowserEnv(): boolean {
+  return typeof window !== 'undefined';
+}
+
+async function getVercelDataDir(): Promise<string | null> {
+  if (isBrowserEnv()) {
+    return '';
+  }
+  const vercelFolder = 'com.vercel.cli';
+  const os = await import('os');
+  const path = await import('path');
+  if (process.env.XDG_DATA_HOME) {
+    return path.join(process.env.XDG_DATA_HOME, vercelFolder);
+  }
+  switch (os.platform()) {
+    case 'darwin':
+      return path.join('~/Library/Application Support', vercelFolder);
+    case 'linux':
+      return path.join('~/.local/share', vercelFolder);
+    case 'win32':
+      return path.join('%LOCALAPPDATA%', vercelFolder);
```

Comment:
The `~` and `%LOCALAPPDATA%` get expanded correctly?

Reply:
Changed these to be expanded properly